collect_app/src/main/java/org/odk/collect/android/utilities/WidgetAppearanceUtils.java

```diff
@@ -0,0 +1,157 @@
+    private static final String EXCEPTION_PARSING_COLUMNS = "Exception parsing columns";
+
+    // Date appearances
+    public static final String ETHIOPIAN        = "ethiopian";
+    public static final String COPTIC           = "coptic";
+    public static final String ISLAMIC          = "islamic";
+    public static final String BIKRAM_SAMBAT    = "bikram-sambat";
+    public static final String MYANMAR          = "myanmar";
+    public static final String NO_CALENDAR      = "no-calendar";
+    public static final String MONTH_YEAR       = "month-year";
+    public static final String YEAR             = "year";
+
+    // Select one/multiple appearances
+    public static final String COMPACT          = "compact";
+    public static final String COMPACT_N        = "compact-";
+    public static final String MINIMAL          = "minimal";
+    public static final String COLUMNS          = "columns";
+    public static final String COLUMNS_N        = "columns-";
+    public static final String COLUMNS_FLEX     = "columns-flex";
+    public static final String QUICKCOMPACT     = "quickcompact";
+    public static final String SEARCH           = "search";
+    public static final String AUTOCOMPLETE     = "autocomplete";
+    public static final String LIST_NO_LABEL    = "list-nolabel";
+    public static final String LIST             = "list";
+    public static final String LABEL            = "label";
+    public static final String IMAGE_MAP        = "image-map";
+    public static final String NO_BUTTONS       = "no-buttons";
+    public static final String QUICK            = "quick";
+
+    // Media appearances
+    public static final String SIGNATURE        = "signature";
+    public static final String ANNOTATE         = "annotate";
+    public static final String DRAW             = "draw";
+    public static final String SELFIE           = "selfie";
+    public static final String NEW_FRONT        = "new-front";
+    public static final String NEW              = "new";
+    public static final String FRONT            = "front";
+
+    // Maps appearances
+    public static final String PLACEMENT_MAP    = "placement-map";
+    public static final String MAPS             = "maps";
+
+    // Other appearances
+    public static final String NO_APPEARANCE    = "";
+    public static final String BEARING          = "bearing";
+    public static final String EX               = "ex:";
+    public static final String THOUSANDS_SEP    = "thousands-sep";
+    public static final String PRINTER          = "printer";
+    public static final String NUMBERS          = "numbers";
+    public static final String URL              = "url";
+    public static final String RATING           = "rating";
+
+    private WidgetAppearanceUtils() {
+    }
+
+    // Get appearance hint and clean it up so it is lower case, without the search function and never null.
+    @NonNull
+    public static String getAppearance(FormEntryPrompt fep) {
```

Comment:
How about putting this method in `FormEntryPrompt` so it can be called on the prompt itself and giving it a name that makes clear that it transforms the appearance in some way? I'm thinking `getSanitizedAppearanceHint` or something like that. When there's a logical place for behavior, it's best to avoid static methods.

Reply:
Oh, right, I just remembered `FormEntryPrompt` is in JavaRosa, not Collect. Sigh.